### PR TITLE
Created the filesystem module.

### PR DESCRIPTION
**Pull request by jpakkane**
I figured we'd start simple. I guess there may be cases where people would want the checks to not follow symlinks, but lt should be rare. Let's add those only after someone actually reports needing them.

### CODE REVIEW

File: mesonbuild/modules/fs.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from . import ExtensionModule
+from . import ModuleReturnValue
+from ..mesonlib import MesonException
+
+from ..interpreterbase import permittedKwargs, stringArgs, noKwargs, FeatureNew, FeatureNewKwargs
```
Comment (ignatenkobrain) [inline mesonbuild/modules/fs.py]: __[Flake8]__ '..interpreterbase.FeatureNew' imported but unused ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6135#issue-4804504))

<sub>posted by Sider</sub>

**Comment by scivision** [comment]
Oh that's great. I had been reluctant to ask for this feature for some reason, so I did it via run_command(python) scripts. But it would be *so* much better to have this in Meson like this!

File: mesonbuild/modules/fs.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from . import ExtensionModule
+from . import ModuleReturnValue
+from ..mesonlib import MesonException
+
+from ..interpreterbase import permittedKwargs, stringArgs, noKwargs, FeatureNewKwargs
```
Comment (ignatenkobrain) [inline mesonbuild/modules/fs.py]: __[Flake8]__ '..interpreterbase.FeatureNewKwargs' imported but unused ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6135#issue-4804505))

<sub>posted by Sider</sub>

File: mesonbuild/modules/fs.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from . import ExtensionModule
+from . import ModuleReturnValue
+from ..mesonlib import MesonException
+
+from ..interpreterbase import permittedKwargs, stringArgs, noKwargs
```
Comment (ignatenkobrain) [inline mesonbuild/modules/fs.py]: __[Flake8]__ '..interpreterbase.permittedKwargs' imported but unused ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6135#issue-4804506))

<sub>posted by Sider</sub>

**Comment by scivision** [comment]
CI: the usual Cygwin symlinks 🤢

**Pull request merged**